operator/main.py

```diff
@@ -650,20 +663,27 @@ def handle_deleted(core_v1_client, obj):
 
         hostvol_type = storage_info_data.get("type")
 
-        # We can't delete external volume but cleanup StorageClass and Configmap
-        # Delete Configmap and Storage class for both Native & External
-        delete_storage_class(volname, hostvol_type)
-        delete_config_map(core_v1_client, obj)
-
         if hostvol_type != "External":
 
-            delete_server_pods(storage_info_data, obj)
             filename = os.path.join(MANIFESTS_DIR, "services.yaml")
-            template(filename, namespace=NAMESPACE, volname=volname)
+            with open(filename, 'r') as f:
+                logging.info(logf(
+                    "services.yaml",
+                    services=f.read()
+                ))
+            resource_to_be_patched = "kadalustorages.kadalu-operator.storage/" + volname
+            lib_execute(KUBECTL_CMD, PATCH_CMD, resource_to_be_patched, "-p",
```

Comment:
- where is the finalizer being set in the first place?

Reply:
The finalizer is being set in storage_config right now through user end[sample-storage.yaml]. I tried to set it by default in crd templates/operator.j2